Testscripts/Linux/STOR-Lis-Disk.sh

```diff
@@ -0,0 +1,186 @@
+    blockSize=$((32 * 1024 * 1024))
+    _gb=$((1 * 1024 * 1024 * 1024))
+    targetSize=$(blockdev --getsize64 $targetDevice)
+    let "blocks=$targetSize / $blockSize"
+
+    if [ "$targetSize" -gt "$_gb" ]; then
+        targetSize=$_gb
+        let "blocks=$targetSize / $blockSize"
+    fi
+
+    blocks=$((blocks - 1))
+    mount $targetDevice /mnt/
+    targetDevice="/mnt/1"
+    LogMsg "Creating test data file $testfile with size $blockSize"
+    LogMsg "We will fill the device $targetDevice (of size $targetSize) with this data (in $blocks) and then will check if the data is not corrupted."
+    LogMsg "This will erase all data in $targetDevice"
+
+    LogMsg "Creating test source file... ($BLOCKSIZE)"
+
+    dd if=/dev/urandom of=$testFile bs=$blockSize count=1 status=noxfer 2>/dev/null
+
+    LogMsg "Calculating source checksum..."
+    checksum=$(sha1sum $testFile | cut -d " " -f 1)
+    LogMsg $checksum
+    LogMsg "Checking ${blocks} blocks"
+    for ((y = 0; y < $blocks; y++)); do
+        LogMsg "Writing block $y to device $targetDevice ..."
+        dd if=$testFile of=$targetDevice bs=$blockSize count=1 seek=$y status=noxfer 2>/dev/null
+        LogMsg "Checking block $y ..."
+        testChecksum=$(dd if=$targetDevice bs=$blockSize count=1 skip=$y status=noxfer 2>/dev/null | sha1sum | cut -d " " -f 1)
+        if [ "$checksum" == "$testChecksum" ]; then
+            LogMsg "Checksum matched for block $y"
+        else
+            LogMsg "Checksum mismatch on  block $y for ${targetDevice} "
+            SetTestStateFailed
+            exit 0
+        fi
+    done
+    LogMsg "Data integrity test on ${blocks} blocks on drive $1 : success "
+    umount /mnt/
+    rm -f $testFile
+}
+
+SetTestStateRunning
+# Count the number of SCSI= and IDE= entries in constants
+#
+diskCount=1
+# Set to 1 because of resource disk
+for entry in $(cat ./constants.sh); do
+    # Convert to lower case
+    lowStr="$(tr '[A-Z]' '[a-z' <<<"$entry")"
+
+    # does it start wtih ide or scsi
+    if [[ $lowStr == ide* ]]; then
+        diskCount=$((diskCount + 1))
+    fi
+
+    if [[ $lowStr == scsi* ]]; then
+        diskCount=$((diskCount + 1))
+    fi
+done
+
+LogMsg "constants disk count = $diskCount"
+
+#
+# Compute the number of sd* drives on the system.
+#
+sdCount=0
+for drive in /dev/sd*[^0-9]; do
+    sdCount=$((sdCount + 1))
+done
+
+#
+# Subtract the boot disk from the sdCount, then make
+# sure the two disk counts match
+#
+sdCount=$((sdCount - 1))
+LogMsg "/dev/sd* disk count = $sdCount"
+
+if [ $sdCount != $diskCount ]; then
+    LogMsg " disk count ($diskCount) does not match disk count from /dev/sd* ($sdCount)"
+    SetTestStateAborted
+    exit 0
+fi
+
+#
+# For each drive, run fdisk -l and extract the drive
+# size in bytes.  The setup script will add Fixed
+#.vhd of size 1GB, and Dynamic .vhd of 137GB
+#
+FixedDiskSize=1073741824
+Disk4KSize=4096
+DynamicDiskSize=136365211648
+
+for driveName in /dev/sd*[^0-9]; do
+    #
```

Comment:
Leave only the comment line and erase just the #

Reply:
Fixed